### ISSUE TITLE
Error Checking for Update - Check network connection

### ISSUE THREAD

**Issue by Zodiac-69** [jisotalo/reciva-radio-patching#1]
I am trying to patch my Reciva radio.
I get to step 10 and i can see the dns queries in the Patcher log, but i am unable to update the firmware.
I get - Error Checking for Update - Check network connection - on the Reviva screen.

One thing i have seen that is "strange" is that when i soft power the radio Off/On, i see: dns: query for 149.2.168.192.in-addr.arpa
This is the Patcher IP in reverse - The Patcher IP is: 192.168.2.149

**Comment by jisotalo** [comment]
Hi!

Have you tried again few times and double-checked all settings? And are you running it as administrator? I haven't faced that issue myself, so it's hard to say what's wrong.

Could you perhaps copypaste the patcher log here?

That "strange" issue is normal as far as I know: https://en.wikipedia.org/wiki/Reverse_DNS_lookup

**Comment by Zodiac-69** [comment]
Hi Jussi

Yes, i run the Patcher as administrator. Patcher DNS set to 8.8.8.8, my Windows 10 PC local IP is 192.168.2.149

The radio i have firmware is as follow:
Service Pack: v257-a-865-a-476
Serial Firmware: 02
Application: v600-a-311-a-679
Helix: v026
Kernel: v24100409+10
Bootloader: v017-c-002

I run the Patcher application and here is the log when the i switch the radio on:
gui: trying to determine IP.
dns: server is running...
gui: servers started...
dns: query for ntp.reciva.com
dns: query for portal1.7803986842.com
dns: query for ntp.reciva.com
dns: query for portal2.7803986842.com
dns: query for portal1.7803986842.com
dns: query for portal3.7803986842.com
dns: query for ntp.reciva.com
dns: query for portal4.7803986842.com
dns: query for portal2.7803986842.com
dns: query for portal5.7803986842.com
dns: query for portal6.7803986842.com
dns: query for portal3.7803986842.com
dns: query for p1.ext.h1.east.us.reciva.com
dns: query for portal7.7803986842.com
dns: query for portal8.7803986842.com
dns: query for portal4.7803986842.com
dns: query for portal9.7803986842.com
dns: query for portal5.7803986842.com
dns: query for portal10.7803986842.com
dns: query for portal6.7803986842.com
dns: query for portal11.7803986842.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for p1.ext.h1.east.us.reciva.com
dns: query for portal7.7803986842.com
dns: query for portal8.7803986842.com
dns: query for portal9.7803986842.com
dns: query for portal10.7803986842.com
dns: query for portal11.7803986842.com
dns: query for portal12.7803986842.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for portal12.7803986842.com
dns: query for portal13.7803986842.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for portal13.7803986842.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for portal14.7803986842.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for portal14.7803986842.com
dns: query for p5-mid-us.reciva.com
dns: query for p1.ext.h3.west.us.reciva.com
dns: query for p1-mid-us.reciva.com
dns: query for portal15.7803986842.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for p5-mid-us.reciva.com
dns: query for p1.ext.h3.west.us.reciva.com
dns: query for p1-mid-us.reciva.com
dns: query for portal15.7803986842.com
dns: query for portal16.7803986842.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for portal16.7803986842.com
dns: query for p1.ext.h1.uk.reciva.com
dns: query for p1.ext.h4.west.us.reciva.com
dns: query for p1.ext.h2.east.us.reciva.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for p1.ext.h1.uk.reciva.com
dns: query for p1.ext.h4.west.us.reciva.com
dns: query for p1.ext.h2.east.us.reciva.com
dns: query for p1.ext.h2.west.us.reciva.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for p1.ext.h2.west.us.reciva.com
dns: query for p4-mid-us.reciva.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for p4-mid-us.reciva.com
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for p2-mid-us.reciva.com
dns: query for p2-mid-us.reciva.com

Once i select "Upgrade Firmware" on my Reciva unit, i only see it query:
dns: query for www.reciva.com
dns: query for www.reciva.com

After this, if i select "Upgrade Firmware" - i have done this many times - i don't see any log entries.

What i am seeing is that every x minutes, the Reviva unit does a rDNS on my local desktop PC IP
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa
dns: query for 149.2.168.192.in-addr.arpa

**Comment by Zodiac-69** [comment]
Hi Jussi

I managed to install the Sharpfin patch.
After leaving the radio on for a long time, it went into a mode where the menu on the radio was very difficult to navigate, slow to respond.
I then ran the Patcher application, but this time i "ticked" the proxy HTTP setting - not sure if this is what made it work?
The moment i selected the firmware update on the radio, i saw "Are you Sure" on the screen.

The fun is to find local stations - i am in South Africa - that still work on the Reciva unit.
Last count i only had three local stations that were still working, the rest i could not get to stream.

Thank you for making it easy to bring my "OXX Digital" Alto reciva powered unit back to life.
